src/qt/updatedialog.cpp

```diff
@@ -1,16 +1,39 @@
 #include <qt/updatedialog.h>
+#include <downloader.h>
+#include <util/system.h>
+#include <boost/algorithm/string.hpp>
+#include <clientversion.h>
 #include <qt/forms/ui_updatedialog.h>
 #include <qt/guiutil.h>
 #include <qt/guiconstants.h>
-#include <downloader.h>
 #include <QDesktopServices>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QProcess>
+#include <QDir>
 
 UpdateDialog::UpdateDialog(QWidget *parent) :
     QDialog(parent),
     ui(new Ui::UpdateDialog)
 {
     ui->setupUi(this);
+    setWindowFlags(Qt::Window | Qt::FramelessWindowHint);
     setWindowTitle(tr("Check for Update"));
+    ui->updateButton->setVisible(false);
+    ui->progressBar->setVisible(false);
+    downloadVersionFile();
+    clientName = getUpdatedClient();
+    if (clientName.compare(" ") != 0){
+
+        ui->updateButton->setVisible(true);
+        std::string stdstr = "Update is available! Click update to download " +clientName;
+        QString qstr = QString::fromStdString(stdstr);
```

Comment:
```suggestion
        QString qstr = QString(QObject::tr("Update is available! Click update to download %1")).arg(clientName);
```
Like that we could translate :)

Reply:
?

Reply:
thought this was committed my bad, it is now